src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -104,6 +107,12 @@ public JsonAdaptedPerson(Person source) {
                 .collect(Collectors.toList()));
     }
 
+    private IllegalValueException internalIllegalValueException(String message) {
+        logger.warning(String.format(MESSAGE_DESERIALIZE_ERROR_DUMP_DATA, "Person"));
+        logger.warning(this.toString());
+        return new IllegalValueException(message);
+    }
```

Comment:
Quite hard to abstract this method because of circular dependencies. I am using `this.toString()` in this method. 

Lets say if I create something like this to abstract:
```java
/**
 * This factory allows the creation of IllegalValueExceptions with inbuilt logging. We assume that everytime
 * a class uses this factory to create an exception, that exception will be thrown and so we log. This factory
 * should only be used for throwing IllegalValueExceptions for errors found during json deserialization.
 */
public class JsonExceptionFactory {
    private final Logger logger;
    private final String clazzName;
    private final Object object;

    public <T> JsonExceptionFactory(Class<T> clazz, String clazzToDeserialize, Object objectToDump) {
        logger = LogsCenter.getLogger(clazz);
        clazzName = clazzToDeserialize;
        object = objectToDump;
    }

    public IllegalValueException build(String message) {
        logger.warning(String.format(MESSAGE_DESERIALIZE_ERROR_DUMP_DATA, clazzName));
        logger.warning(object.toString());
        return new IllegalValueException(message);
    }
}
```

In `JsonAdaptedPerson.java`:
```java
...
private final JsonExceptionFactory exceptionFactory = new
            JsonExceptionFactory(JsonAdaptedPerson.class, "Person", this);
...
throw exceptionFactory.build(String.format(MISSING_FIELD_MESSAGE_FORMAT,
                    Name.class.getSimpleName()));
```

Will have circular dependency because `JsonAdaptedPerson` owns `JsonExceptionFactory` owns `JsonAdaptedPerson`, to put it simply `JsonAdaptedPerson` -> `JsonExceptionFactory` -> `JsonAdaptedPerson`

Reply:
Will further consider abstracting it out if it becomes easier.